### PR TITLE
FIX-#5128: Fix reading parquet directory from s3.

### LINKED ISSUE #5128: BUG: Cannot read parquet file from s3 bucket when using Modin 0.16.1 on Ray
### Modin version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the latest released version of Modin.

- [X] I have confirmed this bug exists on the main branch of Modin. (In order to do this you can follow [this guide](https://modin.readthedocs.io/en/stable/getting_started/installation.html#installing-from-the-github-master-branch).)

### Reproducible Example

```python
import modin.pandas as pd

input_path = 's3://<bucket>/data/'
df = pd.read_parquet(input_path)
```

### Issue Description

When I call `pd.read_parquet('s3://<bucket>/data/')` to read parquet files from s3, it throws error `pyarrow.lib.ArrowInvalid: GetFileInfo() yielded path '<bucket>/data/xxx.snappy.parquet', which is outside base dir 's3://<bucket>/data/'`

### Expected Behavior

When call `pd.read_parquet('s3://<bucket>/data/')`, it should successfully read the data from s3 bucket.

### Error Logs

<details>

```python-traceback

Traceback (most recent call last):
  File "modin_api_test.py", line 195, in <module>
    df = pd.read_parquet(input_path)
  File "/home/ubuntu/.local/lib/python3.8/site-packages/modin/logging/logger_decorator.py", line 128, in run_and_log
    return obj(*args, **kwargs)
  File "/home/ubuntu/.local/lib/python3.8/site-packages/modin/_compat/pandas_api/latest/io.py", line 291, in read_parquet
    query_compiler=FactoryDispatcher.read_parquet(
  File "/home/ubuntu/.local/lib/python3.8/site-packages/modin/core/execution/dispatching/factories/dispatcher.py", line 180, in read_parquet
    return cls.__factory._read_parquet(**kwargs)
  File "/home/ubuntu/.local/lib/python3.8/site-packages/modin/core/execution/dispatching/factories/factories.py", line 207, in _read_parquet
    return cls.io_cls.read_parquet(**kwargs)
  File "/home/ubuntu/.local/lib/python3.8/site-packages/modin/logging/logger_decorator.py", line 128, in run_and_log
    return obj(*args, **kwargs)
  File "/home/ubuntu/.local/lib/python3.8/site-packages/modin/core/io/file_dispatcher.py", line 154, in read
    query_compiler = cls._read(*args, **kwargs)
  File "/home/ubuntu/.local/lib/python3.8/site-packages/modin/logging/logger_decorator.py", line 128, in run_and_log
    return obj(*args, **kwargs)
  File "/home/ubuntu/.local/lib/python3.8/site-packages/modin/core/io/column_stores/parquet_dispatcher.py", line 648, in _read
    return cls.build_query_compiler(dataset, columns, index_columns, **kwargs)
  File "/home/ubuntu/.local/lib/python3.8/site-packages/modin/logging/logger_decorator.py", line 128, in run_and_log
    return obj(*args, **kwargs)
  File "/home/ubuntu/.local/lib/python3.8/site-packages/modin/core/io/column_stores/parquet_dispatcher.py", line 560, in build_query_compiler
    index, sync_index = cls.build_index(dataset, partition_ids, index_columns)
  File "/home/ubuntu/.local/lib/python3.8/site-packages/modin/logging/logger_decorator.py", line 128, in run_and_log
    return obj(*args, **kwargs)
  File "/home/ubuntu/.local/lib/python3.8/site-packages/modin/core/io/column_stores/parquet_dispatcher.py", line 522, in build_index
    complete_index = dataset.to_pandas_dataframe(
  File "/home/ubuntu/.local/lib/python3.8/site-packages/modin/core/io/column_stores/parquet_dispatcher.py", line 226, in to_pandas_dataframe
    return read_table(self.path, columns=columns, filesystem=self.fs).to_pandas()
  File "/home/ubuntu/.local/lib/python3.8/site-packages/pyarrow/parquet/__init__.py", line 2780, in read_table
    dataset = _ParquetDatasetV2(
  File "/home/ubuntu/.local/lib/python3.8/site-packages/pyarrow/parquet/__init__.py", line 2379, in __init__
    self._dataset = ds.dataset(path_or_paths, filesystem=filesystem,
  File "/home/ubuntu/.local/lib/python3.8/site-packages/pyarrow/dataset.py", line 749, in dataset
    return _filesystem_dataset(source, **kwargs)
  File "/home/ubuntu/.local/lib/python3.8/site-packages/pyarrow/dataset.py", line 449, in _filesystem_dataset
    factory = FileSystemDatasetFactory(fs, paths_or_selector, format, options)
  File "pyarrow/_dataset.pyx", line 2059, in pyarrow._dataset.FileSystemDatasetFactory.__init__
  File "pyarrow/error.pxi", line 144, in pyarrow.lib.pyarrow_internal_check_status
  File "pyarrow/error.pxi", line 100, in pyarrow.lib.check_status
pyarrow.lib.ArrowInvalid: GetFileInfo() yielded path '<bucket>/data/xxx.snappy.parquet', which is outside base dir 's3://<bucket>/data/'
```

</details>

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 98a9694694584f50d03f6e27b516eb10d18d36ef
python           : 3.8.10.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.13.0-1025-aws
Version          : #27~20.04.1-Ubuntu SMP Thu May 19 15:17:13 UTC 2022
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : C.UTF-8
LOCALE           : en_US.UTF-8

Modin dependencies
------------------
modin            : 0.16.1
ray              : 2.0.0
dask             : None
distributed      : None
hdk              : None

pandas dependencies
-------------------
pandas           : 1.5.0
numpy            : 1.23.4
pytz             : 2022.4
dateutil         : 2.8.2
setuptools       : 45.2.0
pip              : 20.0.2
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : None
IPython          : 8.5.0
pandas_datareader: None
bs4              : 4.11.1
bottleneck       : None
brotli           : None
fastparquet      : 0.8.3
fsspec           : 2022.8.2
gcsfs            : None
matplotlib       : None
numba            : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : 9.0.0
pyreadstat       : None
pyxlsb           : None
s3fs             : 2022.8.2
scipy            : None
snappy           : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
zstandard        : None
tzdata           : None

</details>

### PR DESCRIPTION
**Pull request by mvashishtha**
Signed-off-by: mvashishtha <mahesh@ponder.io>

## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #5128
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

**Comment by mvashishtha** [comment]
@vnlitvinov I think I've fixed the compat failures.

**Review by vnlitvinov — COMMENTED** [review]
LGTM but got a question

File: modin/pandas/test/test_io.py [hunk 1]
```diff
@@ -1593,6 +1606,31 @@ def test_read_parquet_s3(self, path_type, engine):
...
+                            "older pandas.read_parquet does not "
+                            + "support storage_options, but modin "
+                            + "and pandas both happen to give "
+                            + "different ValueErrors, so we cannot "
+                            + "xfail"
+                        )
+                    )
+            # TODO(mvashishtha): DO NOT MERGE until we make
+            # s3://modin-datasets/testing publicly accessible and test reading
+            # that directory instead.
```
[the remark this answers is not in the document]
Comment (pyrito) [reply modin/pandas/test/test_io.py:1627]: @mvashishtha I thought the bucket is publicly accessible?

**Review by pyrito — COMMENTED** [review]
Apologies, this must have been my mistake for not testing this!

File: modin/pandas/test/test_io.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (pyrito) [reply modin/pandas/test/test_io.py:1627]: Oh, is it that you can't modify it?

[the remark this answers is not in the document]
Comment (mvashishtha) [reply modin/pandas/test/test_io.py:1624]: pandas can't seem to see the directory. if I do:


```python
import pandas
pdf = pandas.read_parquet("s3://modin-datasets/testing/", storage_options=dict(anon=True))
```

I get `FileNotFoundError: modin-datasets/testing/`. @gshimansky is working on a fix [here](https://modin-project.slack.com/archives/CKBB4EVBK/p1665779355185649)

**Review by vnlitvinov — APPROVED** [review]
LGTM assuming tests pass.

**Pull request merged**